Python/ceval.c

```diff
@@ -4603,16 +4635,44 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
 
         TARGET(CALL_FUNCTION): {
             PREDICTED(CALL_FUNCTION);
-            PyObject **sp, *res;
-            sp = stack_pointer;
-            res = call_function(tstate, &sp, oparg, NULL, cframe.use_tracing);
-            stack_pointer = sp;
-            PUSH(res);
-            if (res == NULL) {
+            PyObject *res;
+
+            // Check if the call can be inlined or not
+            PyObject *function = *(stack_pointer - oparg - 1);
+            int inline_call = 0;
+            if (Py_TYPE(function) == &PyFunction_Type) {
+                PyCodeObject *code = (PyCodeObject*)((PyFunctionObject*)(function))->func_code;
+                int is_coro = code->co_flags & (CO_GENERATOR | CO_COROUTINE | CO_ASYNC_GENERATOR);
+                inline_call = (is_coro || cframe.use_tracing) ? 0 : 1;
```

Comment:
Why not ``!(is_coro || cframe.use_tracing)`` ? Seems a little clearer to me. Though you might find the ternary way clearer.

Reply:
I found the ternary operator more explicit so we are not mixing the condition and the data. But i don't feel strongly either way.